template.go

```diff
@@ -0,0 +1,207 @@
+	LocalIPV6Address            string
+	RemoteIPAddress             string
+	RemoteIPV4Address           string
+	RemoteIPV6Address           string
+	Username                    string
+	HTTPMethod                  string
+	HTTPPath                    string
+	HTTPStatusCode              string
+	HTTPRequestSize             string
+	HTTPResponseSize            string
+	HTTPReferer                 string
+	HTTPAgent                   string
+	HTTPVersion                 string
+	HTTPProtocol                string
+	AuthUserID                  string
+	BuildVersion                string
+}
+
+func (g *LogGenerator) Generate() (map[string]string, error) {
+	var (
+		record = make(map[string]string, len(g.logFieldTemplates))
+		data   TemplateData
+		buffer strings.Builder
+	)
+
+	data = TemplateData{
+		Hostname:                    fmt.Sprintf("%s-%d", strings.ToLower(gofakeit.Username()), gofakeit.Number(1, 100)),
+		Priority:                    strconv.Itoa(gofakeit.Number(0, 191)),
+		Application:                 strings.ToLower(gofakeit.Word()),
+		PID:                         strconv.Itoa(gofakeit.Number(1, 10000)),
+		Message:                     gofakeit.HackerPhrase(),
+		NumberZeroToThousand:        strconv.Itoa(gofakeit.Number(0, 1000)),
+		NumberZeroToTenThousand:     strconv.Itoa(gofakeit.Number(0, 10000)),
+		NumberZeroToHundredThousand: strconv.Itoa(gofakeit.Number(0, 100000)),
+		NumberZeroToMillion:         strconv.Itoa(gofakeit.Number(0, 1000000)),
+		IPAddress:                   gofakeit.IPv4Address(),
+		IPV4Address:                 gofakeit.IPv4Address(),
+		IPV6Address:                 gofakeit.IPv6Address(),
+		LocalIPAddress:              gofakeit.IPv4Address(),
+		LocalIPV4Address:            gofakeit.IPv4Address(),
+		LocalIPV6Address:            gofakeit.IPv6Address(),
+		RemoteIPAddress:             gofakeit.IPv4Address(),
+		RemoteIPV4Address:           gofakeit.IPv4Address(),
+		RemoteIPV6Address:           gofakeit.IPv6Address(),
+		Username:                    strings.ToLower(gofakeit.Username()),
+		HTTPMethod:                  gofakeit.HTTPMethod(),
+		HTTPPath:                    randResourceURI(),
+		HTTPStatusCode:              strconv.Itoa(gofakeit.StatusCode()),
+		HTTPRequestSize:             strconv.Itoa(gofakeit.Number(1000, 10000)),
+		HTTPResponseSize:            strconv.Itoa(gofakeit.Number(1000, 10000)),
+		HTTPReferer:                 fmt.Sprintf("https://%s%s", gofakeit.DomainName(), randResourceURI()),
+		HTTPAgent:                   randUserAgent(),
+		HTTPVersion:                 randHTTPVersion(),
+		HTTPProtocol:                randHTTPVersion(),
+		AuthUserID:                  randAuthUserID(),
+		BuildVersion:                randBuildVersion(),
+	}
+
+	for k, v := range g.logFieldTemplates {
+		buffer.Reset()
+		if err := v.Execute(&buffer, data); err != nil {
+			return nil, fmt.Errorf("failed to execute template for field '%s': %w", k, err)
+		}
+		record[k] = buffer.String()
```

Comment:
🤔
This will always produce a string. Not allowing, numbers, booleans or more complex data types.